ui/app/src/modules/Preview/Tools/AudiencesPanel.tsx

```diff
@@ -171,19 +172,19 @@ export default function AudiencesPanel() {
 
   // TODO: Update to new state structure
   const saveProfile = () => {
-    // let params = encodeURI(Object.entries(currentProfile).map(([key, val]) => `${key}=${val}`).join('&'));
-    //
-    // get(`/api/1/profile/set?${params}`).subscribe(
-    //   ({ response }) => {
-    //     dispatch(setAudiencesPanelProfile(site, currentProfile));
-    //     // TODO: display success message
-    //   }
-    // );
+    const model = reversePluckProps(state.model, 'craftercms');
+    const params = encodeURI(Object.entries(model).map(([key, val]) => `${key}=${val.key}`).join('&'));
+
+    get(`/api/1/profile/set?${params}`).subscribe(
+      ({ response }) => {
+        // TODO: display success message
+      }
+    );
   };
 
   const setDefaults = (config) => {
     Object.keys(config.fields).forEach((property: any) => {
```

Comment:
Set them all at once; in a single dispatch.

Both the set profile above and this op is the same service call. Move to an epic.